### PR TITLE
feat: add bls public keys to commit transactions and mined blocks

### LINKED ISSUE #1229: Include BLS public key in commitments and Block Proposal
Nodes should inform other nodes about their BLS public key through commitment and block proposal messages. This field should be optional.

### PR DESCRIPTION
**Pull request by mariocao**
Closes #1229 

This PR:

- updates protobuf schema
- updates data structures (commit transaction body and block header)
- initializes the chain manager with the bn256 public key (from signature manager)
- includes the secp bls mapping (@tmpolaczyk)
- adds protobuf convert for Option<T>

### CODE REVIEW

File: node/src/actors/chain_manager/mining.rs [hunk 1]
```diff
@@ -253,6 +254,9 @@ impl ChainManager {
         let tx_pending_timeout = self.tx_pending_timeout;
         let timestamp = u64::try_from(get_timestamp()).unwrap();
 
+        // FIXME: Check if node is not in ARS to send bn256 public key or None
+        let is_ars_member = false; //act.chain_state.reputation_engine.is_member(own_pkh)
```
Comment (girazoki) [inline node/src/actors/chain_manager/mining.rs:258]: act.chain_state.reputation_engine.ars().contains(&own_pkh)

File: data_structures/src/proto/mod.rs [hunk 2]
```diff
@@ -195,6 +195,27 @@ impl ProtobufConvert for [u8; 32] {
...
+    }
+    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, Error> {
+        let res = T::from_pb(pb)?;
+        if res == T::default() {
+            Ok(Some(res))
+        } else {
+            Ok(None)
+        }
+    }
+}
```
Comment (aesedepece) [inline data_structures/src/proto/mod.rs:217]: Gold :ok_hand:

File: node/src/actors/chain_manager/mod.rs [hunk 3]
```diff
@@ -1059,6 +1072,21 @@ fn update_reputation(
         log::error!("Error updating reputation in consolidation: {}", e);
     }
 
+    secp_bls_mapping.retain(|k| {
+        // Retain identities that exist in the ARS
+        if rep_eng.ars().contains(k) {
```
Comment (girazoki) [inline node/src/actors/chain_manager/mod.rs:1072]:
```suggestion
        if rep_eng.is_ars_member(k){
```

File: data_structures/src/chain.rs [hunk 4]
```diff
@@ -2437,6 +2440,20 @@ impl AltKeys {
...
+            self.insert_bn256(block.block_header.proof.proof.pkh(), value);
+        }
+
+        // Add bn256 public keys from commitment transactions
+        for commit in &block.txns.commit_txns {
+            if let Some(value) = commit.body.bn256_public_key.clone() {
+                self.insert_bn256(commit.body.proof.proof.pkh(), value);
+            }
+        }
+    }
```
Comment (aesedepece) [inline data_structures/src/chain.rs:2456]: Can we document this function?

File: data_structures/src/chain.rs [hunk 5]
```diff
@@ -2558,6 +2575,10 @@ impl ReputationEngine {
     pub fn clear_threshold_cache(&self) {
         self.threshold_cache.borrow_mut().clear_threshold_cache()
     }
+
+    pub fn is_ars_member(&self, pkh: &PublicKeyHash) -> bool {
+        self.ars.contains(pkh)
+    }
```
Comment (aesedepece) [inline data_structures/src/chain.rs:2581]: Can we document this function?

File: node/src/actors/chain_manager/mod.rs [hunk 6]
```diff
@@ -1059,6 +1067,21 @@ fn update_reputation(
...
     }
 
+    secp_bls_mapping.retain(|k| {
+        // Retain identities that exist in the ARS
+        if rep_eng.is_ars_member(k) {
+            // Keep identity
+            true
+        } else {
+            // Remove identity
+            log::trace!(
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs:1077]: Remove this log
```suggestion
```

File: node/src/actors/chain_manager/mod.rs [hunk 7]
```diff
@@ -58,12 +58,13 @@ use crate::{
     },
     signature_mngr, storage_mngr,
 };
+use witnet_data_structures::chain::AltKeys;
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs:61]: Merge this import with the others

**Pull request merged**